core_lib/data_layers/data/handler/sql_alchemy_data_handler_registry.py

```diff
@@ -29,13 +27,12 @@ def connection(self):
 
     def get(self, use_parent_instance=False, *args, **kwargs) -> SqlAlchemyDataHandler:
         if use_parent_instance:
-            db_session = self.instance_under_path.get()
+            db_session = object
             if db_session:
                 db_session_count = self.session_to_count[db_session]
                 self.session_to_count[db_session] = db_session_count + 1
             else:
                 db_session = SqlAlchemyDataHandler(self._engine, use_parent_instance, self._on_db_session_exit)
-                self.instance_under_path.store(db_session)
                 self.session_to_count[db_session] = 1
             return db_session
         else:

```

Comment:
in `def _on_db_session_exit(self, db_session: SqlAlchemyDataHandler):`
remove completely the referenct to `if db_session.use_parent_instance:`

or any other references to `use_parent_instance:`